plugin/vcs/gitlab/gitlab.go

```diff
@@ -233,6 +249,46 @@ func (provider *Provider) TryLogin(ctx context.Context, oauthCtx common.OauthCon
 	return fetchUserInfo(ctx, oauthCtx, instanceURL, "user")
 }
 
+// FetchCommitByID fetch the commit data by id.
+func (provider *Provider) FetchCommitByID(ctx context.Context, oauthCtx common.OauthContext, instanceURL, repositoryID, commitID string) (*vcs.Commit, error) {
+	code, body, err := httpGet(
+		instanceURL,
+		fmt.Sprintf("projects/%s/repository/commits/%s", repositoryID, commitID),
+		&oauthCtx.AccessToken,
+		oauthContext{
+			ClientID:     oauthCtx.ClientID,
+			ClientSecret: oauthCtx.ClientSecret,
+			RefreshToken: oauthCtx.RefreshToken,
+		},
+		oauthCtx.Refresher,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch commit data on GitLab instance %s, err: %w", instanceURL, err)
+	}
+	if code >= 300 {
+		return nil, fmt.Errorf("failed to fetch commit data on GitLab instance %s, status code: %d",
+			instanceURL,
+			code,
+		)
+	}
+
+	commit := &Commit{}
+	if err := json.Unmarshal([]byte(body), commit); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal commit data from GitLab instance %s, err: %w", instanceURL, err)
+	}
+
+	createdTime, err := time.Parse(time.RFC3339Nano, commit.CreatedAt)
```

Comment:
Would RFC3339 be enough or GitLab API requires Nano?

Reply:
Yes, it's the raw data from GitLab.
```json
"committed_date": "2012-09-20T11:50:22+03:00",
```

Reply:
This looks like RFC3339 should be enough instead of RFC3339Nano which is nanosecond precision?